source/beamline/BioXAS/BioXASSideM1MirrorBendControl.cpp

```diff
@@ -0,0 +1,42 @@
+#include "BioXASSideM1MirrorBendControl.h"
+
+BioXASSideM1MirrorBendControl::BioXASSideM1MirrorBendControl(const QString &name, const QString &units, QObject *parent, const QString &description) :
+	BioXASMirrorBendControl(name, units, parent, description)
+{
+	// Initialize inherited variables.
+
+	setMinimumValue( calculateBendRadius(12, 12) );
+	setMaximumValue( calculateBendRadius(0.0001, 0.0001) );
+	setTolerance(100);
+
+	// Current settings.
+
+	updateStates();
+}
+
+BioXASSideM1MirrorBendControl::~BioXASSideM1MirrorBendControl()
+{
+
+}
+
+double BioXASSideM1MirrorBendControl::calculateBendRadius(double upstreamBenderValue, double downstreamBenderValue)
+{
+	double radius1 = -18654.3 + 144742/upstreamBenderValue + 5413.6 * log(upstreamBenderValue);
```

Comment:
One small question, are the explicit numbers the result of a calculation?  Are we worried about precision errors?